### PR TITLE
[Fix] schedulers with no max_steps param

### PR DESCRIPTION
**Pull request by stevehuang52**
All LR schedulers in PyTorch do not have the `max_steps` parameter, so we should not add `max_steps` to their `scheduler_args`. Previous code tackle the problem in case-by-case manner, while here we try to solve the problem in a more general way. 

To do so, we define a set of schedulers that don't have the `max_steps` parameter, and set the `add_max_args_flag` accordingly. 

Similarly, we define a set of epoch-based schedulers to set the `interval`, while previous code only checks for `ReduceLROnPlateau` scheduler.

Besides, the previous code also has bugs, since it doesn't set the correct `add_max_args_flag` and `interval` when `if 'args' in scheduler_config` is `True`.

### CODE REVIEW

File: nemo/core/optim/lr_scheduler.py [hunk 1]
```diff
@@ -681,18 +681,19 @@ def prepare_lr_scheduler(
...
-                interval = 'epoch'
-
             scheduler_args.pop('name', None)
             scheduler_args.pop('t_max_epochs', None)
             scheduler_args.pop('t_accumulate_grad_batches', None)
             scheduler_args.pop('t_limit_train_batches', None)
             scheduler_args.pop('t_num_workers', None)
             scheduler_args.pop('monitor', None)
             scheduler_args.pop('reduce_on_plateau', None)
+
```
Comment (titu1994) [inline nemo/core/optim/lr_scheduler.py:691]: Avoid this and just do use python inspect module to check if the scheduler needs max_steps or not inside of get scheduler. Much simpler logic.

**Review by titu1994 — CHANGES_REQUESTED** [review]

File: nemo/core/optim/lr_scheduler.py (hunk 1 above)
Comment (stevehuang52) [reply nemo/core/optim/lr_scheduler.py]: Thanks for the suggestion, I have changed to using `inspect.signature()` instead. By the way, why do we pass the `scheduler_args` twice, in `scheduler_cls = get_scheduler(scheduler_name, **scheduler_args)` and `schedule = scheduler_cls(optimizer, **scheduler_args)`?

Comment (titu1994) [reply nemo/core/optim/lr_scheduler.py]: One is to get a class, another is to instantiate the class with those kwargs

**Review by titu1994 — APPROVED** [review]
Lgtm

**Review by titu1994 — APPROVED** [review]

**Comment by titu1994** [comment]
@ericharper  for approval

**Comment by evilc3** [comment]
Hey so with this pr the I don't need to define max_step param for any scheduler ?

**Comment by stevehuang52** [comment]
> Hey so with this pr the I don't need to define max_step param for any scheduler ?

Not really, this PR aims to fix the bug that current code will add a max_steps param to some pytorch schedulers that don't require max_steps. 

For those schedulers that do have a max_steps param, current NeMo code will calculate a max_steps if it's not explicitly specified by the user.
